src/AeroSharp/DataAccess/MapConfiguration.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Aerospike.Client;
+using System.Diagnostics.CodeAnalysis;
+
+namespace AeroSharp.DataAccess;
+
+/// <summary>
+///     Configuration that will help build a MapPolicy for Aerospike Map operations.
+///     Reference: https://www.aerospike.com/apidocs/csharp/html/T_Aerospike_Client_MapWriteFlags.htm
+/// </summary>
+[ExcludeFromCodeCoverage]
+public sealed class MapConfiguration
+{
+    public MapConfiguration()
+    {
+        Partial = false;
+        NoFail = false;
+        CreateOnly = false;
+        UpdateOnly = false;
+        Order = MapOrder.UNORDERED;
+    }
+
+    /// <summary>
+    ///     Allow other valid map items to be committed if a map item is denied due to write flag constraints.
+    /// </summary>
+    public bool Partial { get; set; }
```

Comment:
The naming of `Partial` here is a little ambiguous. I see in the comment it's for writes/commits, but it's not immediately obvious in reading the property.

Could we do something like `AllowPartialWrites` or `AllowPartialCommits`? (Potentially also for `NoFail`?)

Reply:
Yeah, to be fair, Aerospike themselves doesn't make this super clear either on their end: https://github.com/aerospike/aerospike-client-csharp/blob/80f15b7ffa44af30a8f4e1eeb0a28a1e7cada9a5/AerospikeClient/CDT/MapWriteFlags.cs

But I agree with @farmerau here - definitely would be good to make this a bit clearer for clients when setting it on their end.

Reply:
Adjusted this to `AllowPartial` for now. Open to other names if you think there's something better we can use.